warehouse/forklift/legacy.py

```diff
@@ -1208,10 +1208,20 @@ def file_upload(request):
     # Ensure the filename doesn't contain any characters that are too 🌶️spicy🥵
     _validate_filename(filename)
 
+    # Extract the project name from the filename and normalize it.
+    filename_prefix = pkg_resources.safe_name(
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏this is using the deprecated `pkg_resources` package - should we take the time now to replace, or do you want for forge ahead and let this get bundled into #13991 (or another)?

Reply:
Yeah, I'd like it to get bunded into the fix for #7811, this is sort of orthogonal.